### ISSUE TITLE
Issues with updating Axis.tick_labels

### ISSUE THREAD

**Issue by astrofrog** [bqplot/bqplot#1536]
Updating tick labels using ``Axis.tick_labels`` doesn't always work properly - here are cases that currently don't work well:

* If I run all cells in the notebook 'Axis Properties.ipynb' in one go the labels do not update to be roman numerals, but things work fine if I run slowly cell by cell, which seems to suggest some kind of race condition if the labels are set before the figure is visible.
* Similarly, if I set the tick values and formats in the same cell, sometimes this doesn't work, which is probably the same as the above issue.
* If the tick values are updated, the tick labels revert to not having custom labels even if the new tick labels have entries in the tick_labels dictionary - for example if the tick_labels dictionary has entries for tick values of [1, 2, 3] and I change the tick_values from [1, 2, 3] to [1, 2] then the tick labels are not used. It's then not possible to re-assign tick_labels to the same dictionary as before since it won't trigger any updates so one needs to change it to an empty dictionary and change it back.

**Comment by maartenbreddels** [comment]
Yes, it's a feature in bqplot that was broken that Tom needed for custom labeling of axes.

**Comment by pllim** [comment]
And what does Jdaviz need custom labeling for again? Sorry, I just need to see if we have tickets need updating downstream, as I don't see any linked here.

**Comment by maartenbreddels** [comment]
I remember this having to do with wavelength vs pixels, which are non-linear iirc I think @kecnry knows best.

**Comment by kecnry** [comment]
yes, this was for the non-linear axes scaling coming to glue-jupyter soon (hopefully)!  Its great to see this unblocked and moving forward, thanks @maartenbreddels and @astrofrog
